core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java

```diff
@@ -861,10 +849,18 @@ public void startCheckerThread() {
   }
 
   void closeExpiredConnections() {
-    List<CachedConnection> expiredConnections;
-
+    ArrayList<CachedConnection> expiredConnections = new ArrayList<>();
     ConnectionPool pool = getConnectionPool();
-    expiredConnections = pool.removeExpiredConnections(killTime);
+
+    for (Entry<ThriftTransportKey,CachedConnections> entry : pool.connections.entrySet()) {
+      Iterator<CachedConnection> iter = entry.getValue().unreserved.iterator();
+      while (iter.hasNext()) {
+        pool.executeWithinLock(entry.getKey(), (key) -> {
+          entry.getValue().removeExpiredConnections(expiredConnections, killTime);
+          entry.getValue().checkReservedForStuckIO();
+        });
+      }
+    }
```

Comment:
I don't understand why this is iterating over the `pool.connections.entrySet()`, but is checking for a non-empty iterator from `entry.getValue().unreserved`. Is the `iter.hasNext()` even necessary? The iterator doesn't seem to be advanced by calling `next()` ever. Is that check performing something important?

Reply:
This was similar to how it was handled before the refactoring. Without the Iter, the thread never gets picked up and removed.

Reply:
There are a couple of other loop options that seem to work but none that are just the for loop.

Reply:
I was having a hard time understanding this change.   I attempted to fix this issue in #1477

Reply:
I will take a look at your changes. My goal was to fix the bug without reverting too much of the refactor that was done to it. In an older release, a similar method was done.

Reply:
I will take a look at your changes. My goal was to fix the bug without reverting too much of the refactor that was done to it. In an older release, a similar method was done.